### PR TITLE
Implement ctrlc in place of signal hook on Windows

### PR DESCRIPTION
**Pull request by seriouscatsoserious**
Satisfies #709, utilizing ctrlc in place of signal hook on Windows only.

### CODE REVIEW

**Comment by romanz** [comment]
Many thanks for the contribution!

**Review by romanz — CHANGES_REQUESTED** [review]
Would it please be possible to add a CI job to make sure the new code builds as expected on Windows?

**Comment by seriouscatsoserious** [comment]
Should note that there is no Windows job with --features metrics_process flag because Prometheus doesn't work on Windows and I didn't want to introduce more complexity just for that

File: Cargo.toml [hunk 1]
```diff
@@ -36,9 +36,14 @@ rayon = "1.9"
 serde = "1.0"
 serde_derive = "1.0, <=1.0.171"  # avoid precompiled binaries (https://github.com/serde-rs/serde/issues/2538)
 serde_json = "1.0"
-signal-hook = "0.3"
 tiny_http = { version = "0.12", optional = true }
 
+[target.'cfg(windows)'.dependencies]
+ctrlc = "3.2.5"
```
Comment (romanz) [inline Cargo.toml:42]: Just from curiosity, why not using 3.4.2 `ctrlc`?
IIUC it should support our MSRV: https://github.com/Detegr/rust-ctrlc/blob/3.4.2/Cargo.toml#L15

**Comment by seriouscatsoserious** [comment]
Updated to 3.4.2

File: rust-toolchain.toml [hunk 2]
```diff
@@ -1,3 +1,3 @@
 [toolchain]
-channel = "1.63.0"
+channel = "1.69.0"
```
Comment (romanz) [inline rust-toolchain.toml:2]: I would prefer to keep the MSRV to 1.63, to allow building this project using Debian stable (https://packages.debian.org/stable/rust/rustc).

Comment (seriouscatsoserious) [reply rust-toolchain.toml]: Got it, done!

**Review by romanz — APPROVED** [review]
Many thanks!

**Pull request merged**